### PR TITLE
Add search functionality

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Add search functionality by Google Custom Search Engine
- Add search functionality by Algolia

**Related issue(s)**
https://github.com/kyma-project/website/issues/307
https://github.com/kyma-project/website/issues/308
Epic: https://github.com/kyma-project/website/issues/18

### CODE REVIEW

File: src/components/search/Algolia.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import { ALGOLIA } from "@common/constants";
+
+import AlgoliaWrapper from "./AlgoliaWrapper";
+
+const Algolia: React.FunctionComponent = () => {
+  const [initial, setInitial] = useState<boolean>(false);
+  const [loadedAlgolia, setLoadedAlgolia] = useState<boolean>(false);
+  const [focused, setFocused] = useState<boolean>(false);
+
+  const autocompleteSelected = (input: any, event: any) => {
```
Comment (aerfio) [inline src/components/search/Algolia.tsx]: https://github.com/typescript-cheatsheets/react-typescript-cheatsheet#forms-and-events

please do not put `any` in places where types get a bit harder

File: src/components/search/Algolia.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    // creating a tag for easier extracting parts of url
+    const a = document.createElement(`a`);
```
Comment (aerfio) [inline src/components/search/Algolia.tsx]: I don't like this, feels way to hacky, use normal object

File: src/components/search/AlgoliaWrapper.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
+import styled, { media } from "@styled";
+import { customScrollBar } from "@styled/mixins";
+
+interface Props {
```
Comment (aerfio) [inline src/components/search/AlgoliaWrapper.ts]: `AlgoliaProps`

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1 @@
+/static/**/*.min.js
```
Comment (aerfio) [inline .eslintignore]: We do not use eslint

File: src/components/layout/SiteMetadata.tsx [hunk 5]
```diff
@@ -134,6 +134,12 @@ const SiteMetadata: React.FunctionComponent<MetadataProps & IntlInterface> = ({
           href: withPrefix("/favicon-16x16.png"),
         },
       ]}
+      script={[
```
Comment (aerfio) [inline src/components/layout/SiteMetadata.tsx]: Either this or `/static/js/docsearch.min.js`

File: src/components/search/Algolia.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        init();
+      });
+    } else {
+      init();
+    }
+  };
+
+  useEffect(() => {
+    mount();
+  });
```
Comment (aerfio) [inline src/components/search/Algolia.tsx]: This triggers every update

```suggestion
  },[]);
```

File: src/components/search/Algolia.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+import Icon from "@components/shared/Icon";
+
+import { ALGOLIA } from "@common/constants";
+
+import AlgoliaWrapper from "./AlgoliaWrapper";
+
+const Algolia: React.FunctionComponent = () => {
+  const [initial, setInitial] = useState<boolean>(false);
+  const [loadedAlgolia, setLoadedAlgolia] = useState<boolean>(false);
+  const [focused, setFocused] = useState<boolean>(false);
```
Comment (aerfio) [inline src/components/search/Algolia.tsx]: Maybe using `useReducer` would be more beneficial here? Not a must, but a nice thing to have

File: src/components/search/Algolia.tsx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+  const init = (): void => {
+    if (initial || !window || !(window as any).docsearch) return;
+
+    (window as any).docsearch(algoliaOptions);
+    setInitial(true);
+  };
+
+  const loadJS = () => import(`@static/js/docsearch.min.js`);
+  const mount = (): void => {
```
Comment (aerfio) [inline src/components/search/Algolia.tsx]: maybe sth like `mountAlgolia` or `loadAlgolia(Script)`

File: .eslintignore (hunk 4 above)
Comment (magicmatatjahu) [reply .eslintignore]: But gatsby use, and I had a problem with `gatsby build` with 3 party script in static folder - `docsearch.min.js`

File: src/components/search/AlgoliaWrapper.ts (hunk 3 above)
Comment (magicmatatjahu) [reply src/components/search/AlgoliaWrapper.ts]: I think that Props is better name if we have only one Props interface in file.

File: src/components/search/Algolia.tsx (hunk 2 above)
Comment (magicmatatjahu) [reply src/components/search/Algolia.tsx]: No

File: .eslintignore (hunk 4 above)
Comment (aerfio) [reply .eslintignore]: ok then

File: src/components/search/Algolia.tsx (hunk 2 above)
Comment (aerfio) [reply src/components/search/Algolia.tsx]: yes

File: src/components/search/AlgoliaWrapper.ts (hunk 3 above)
Comment (aerfio) [reply src/components/search/AlgoliaWrapper.ts]: we have one props _for now_

File: src/components/search/Algolia.tsx (hunk 2 above)
Comment (magicmatatjahu) [reply src/components/search/Algolia.tsx]: No

**Pull request merged**